src/dos/dos.cpp

```diff
@@ -504,13 +504,15 @@ static Bitu DOS_21Handler(void) {
 		break;
 	}
 	case 0x2d:		/* Set System Time */
-		LOG(LOG_DOSMISC,LOG_ERROR)("DOS:Set System Time not supported");
-		//Check input parameters nonetheless
 		if( reg_ch > 23 || reg_cl > 59 || reg_dh > 59 || reg_dl > 99 )
 			reg_al = 0xff; 
 		else { //Allow time to be set to zero. Restore the orginal time for all other parameters. (QuickBasic)
 			if (reg_cx == 0 && reg_dx == 0) {time_start = mem_readd(BIOS_TIMER);LOG_MSG("Warning: game messes with DOS time!");}
 			else time_start = 0;
+			uint32_t ticks = (uint32_t)(
+			        ((double)(reg_ch * 3600 + reg_cl * 60 + reg_dh)) *
+			        18.206481481);
+			mem_writed(BIOS_TIMER, ticks);
```

Comment:
I wonder if we had the 18.2~ ticks/second defined as a constant somewhere - but turns out we have the equivalent in ticks-per-day (1573040) used in the `TIME.COM` command:

https://github.com/dosbox-staging/dosbox-staging/blob/main/src/shell/shell_cmds.cpp#L1428-L1451


![2021-11-18_09-54](https://user-images.githubusercontent.com/1557255/142470856-b3880329-1d70-4274-9ca4-7e113d3d72e0.png)

Maybe the snippet that converts hours, minutes, seconds to a uint32_t tick-count could be lifted out into timer.h/cpp, and then called in both places?

Reply:
I agree with the suggestion that it is better for the constant to appear only once. By re-looking at the code of both DATE and TIME commands, it appears that for the former (DATE), both manual and host date synchronization calls the 0x2d function in dos.cpp, so I think we can probably do the same for TIME command, i.e. move the constant (1573040) to the 0x2d function in dos.cpp, and make both manual and host time synchronization calls this function. By doing so, the handlings of both DATE and TIME commands are now consistent, and the constant only appears once in the source code.

Reply:
Thanks for the really nice approach, @Wengier! Now only one chunk of the code converts time to ticks,  and everything else calls it. Doesn' get [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself)'er than that!